pixeltable/tests/test_client.py

```diff
@@ -3,6 +3,84 @@
 import pixeltable as pt
 from pixeltable import exceptions as exc
 
+import pyarrow.parquet as pq
+import pyarrow as pa
+from pixeltable.utils.arrow import iter_tuples
+
+import datasets
+from collections import namedtuple
+
+from datetime import datetime
+from pathlib import Path
+
+def make_test_arrow_table(output_path : Path):
+    value_dict = {
+        'c_id' : [1,2,3,4,5],
+        'c_int64': [-10,-20,-30,-40, None],
+        'c_int32': [-1,-2,-3,-4,None],
+        'c_uint64': [10,20,30,40, None],
+        'c_uint32': [1,2,3,4, None],
+        'c_float32': [1.1,2.2,3.3,4.4, None],
+        'c_string': ['aaa','bbb','ccc','ddd', None],
+        'c_boolean': [True,False,True,False, None],
+        'c_timestamp': [datetime(2012, 1, 1, 12, 0, 0, 25),
+                        datetime(2012, 1, 2, 12, 0, 0, 25),
+                        datetime(2012, 1, 3, 12, 0, 0, 25),
+                        datetime(2012, 1, 4, 12, 0, 0, 25),
+                        None
+                        ],
+        'c_array_float32': [[1., 2.,], [10., 20.,], [100., 200.,], [1000., 2000.,], [10000., 20000.]],
+    }
+
+    arr_size = len(value_dict['c_array_float32'][0])
+    tensor_type = pa.fixed_shape_tensor(pa.float32(), (arr_size,))
+    storage =  pa.array(value_dict['c_array_float32'], pa.list_(pa.float32(), arr_size))
+    extension_arr = pa.ExtensionArray.from_storage(tensor_type, storage)
+    value_dict['c_array_float32'] = extension_arr
+
+    # The pyarrow fixed_shape_tensor type supports nulls during save, but fails to read them back,
+    # so it is unclear if any parquet we read will can have nulls in the tensor column.
+    schema=pa.schema([
+        ('c_id', pa.int32()),
+        ('c_int64', pa.int64()),
+        ('c_int32', pa.int32()),
+        ('c_uint64', pa.uint64()),
+        ('c_uint32', pa.uint32()),
+        ('c_float32', pa.float32()),
+        ('c_string', pa.string()),
+        ('c_boolean', pa.bool_()),
+        ('c_timestamp', pa.timestamp('us')),
+        ('c_array_float32', tensor_type),
+    ])
+
+    test_table = pa.Table.from_pydict(value_dict, schema=schema)
+    pq.write_table(test_table, str(output_path / 'test.parquet'))
+
+def _check_values_equal(hf_dataset : datasets.Dataset, df : pt.DataFrame, split_column_name : str):
```

Comment:
should also go into test/utils.py. see assert_resultset_eq

also missing return type